yt/fields/field_info_container.py

```diff
@@ -469,8 +471,9 @@ def check_derived_fields(self, fields_to_check=None):
         for field in fields_to_check:
             fi = self[field]
             try:
+                # fd: field detector
                 fd = fi.get_dependencies(ds=self.ds)
-            except Exception as e:
+            except (YTFieldNotFound, NeedsConfiguration, UnitConversionError) as e:
```

Comment:
this is not strictly needed here, but as a side effect of exploring this function I figured I might as well catch relevant exceptions only. If I've missed any, tests should tell !